torch/_utils_internal.py

```diff
@@ -155,6 +155,98 @@ def export_api_rollout_check() -> bool:
     return False
 
 
+class JustknobsConfig:
+    """Represents a lazily loaded config
+
+    This is designed to be used to specify a value in a config.
+
+    i.e. foo.bar = JustknobsConfig("//foo:bar", env_name="FORCE_FOO_BAR")
+
+    This will lazily evaluate it's configuration when compared
+    i.e. if foo.bar:
+
+    It can also be assigned to - i.e.
+    foo.bar = JustknobsConfig("//foo:bar")
+    foo.bar = True
+
+    However, this will force it to always be the overriden value. There isn't a easy way to fix this within
+    python, given how _config.py is generally used.
+
+    Note that this does allow for no JK name (so that you can use this to replace old configurations).
+    """
+
+    def __init__(self, name: Optional[str] = None, env_name=None, default: bool = True):
+        self.name = name
+        self.env_name = env_name
+        self.default = default
+
+    def __bool__(self):
+        return justknobs_feature(
+            self.name, env_name=self.env_name, default=self.default
+        )
+
+    def __str__(self):
+        v = bool(self)
+        return f"JustknobsConfig(name={self.name}, env_name={self.env_name}, default={self.default} - evals_to={v})"
+
+
+def justknobs_feature(
+    name: Optional[str], config_value=None, env_name=None, default: bool = True
+):
+    """Returns whether or not a specific justknob feature is enabled.
+
+    This is a slightly higher level API then justknobs_check, designed to make it "easy" to do the right thing.
+    The primary thing it does, is allow configuration to override JK by default, while retaining some features to force this
+    the other way during sevs.
+
+    The preference order (i.e. who wins first) in OSS (and FB) is
+    - Environment Variable if specified
+    - Config is specified
```

Comment:
This ordering is inconsistent with the de facto current ordering, which is that explicit config specification in code overrides envvar. I guess you wanted to swap it so that we could use envvars to force behavior change even if a model had explicitly configured one way, but it's probably more important to be BC (unless we do some sort of data collection to show that it's OK not to break it.)

Reply:
Flipped to be consistent. I'll probably do some clever stuff internally to make it possible to switch this ordering around.